sdk/dns/Azure.ResourceManager.Dns/tests/Helper.cs

```diff
@@ -22,25 +22,20 @@ public static async Task TryRegisterResourceGroupAsync(ResourceGroupsOperations
             Azure.ResourceManager.Dns.Models.Resource second,
             bool ignoreEtag = false)
         {
-            if (first == null && second == null)
-            {
-                return true;
-            }
-            else if (first == null || second == null)
+
+            if (first == null || second == null)
             {
                 return false;
             }
 
-            if (first.Location != second.Location
-                || first.Name != second.Name)
+            if (first.Location.ToLower() != second.Location.ToLower())
             {
                 return false;
             }
 
-            if (first.Tags != null || second.Tags != null)
+            if (first.Tags != null && second.Tags != null)
             {
-                if (first.Tags == null || second.Tags == null ||
-                    first.Tags.Count != second.Tags.Count)
+                if (first.Tags.Count != second.Tags.Count)
```

Comment:
nit: if you ever need to compare content of Tags collection, LINQ's Intersect, Except are useful.